### PR TITLE
PY2 & PY3 String/Binary Fixes

### PR DESCRIPTION
**Pull request by JVickery-TBS**
fix(misc): fixed binary/str types between py2 and py3;

- Added binary type support and mapping.
- Added new config option for strict type guessing.
- Minor fixes for python2 types and classes.
- Minor fix for failed type guessing.

### CODE REVIEW

File: ckanext/xloader/config_declaration.yaml [hunk 1]
```diff
@@ -48,6 +48,14 @@ groups:
         type: bool
         required: false
         legacy_key: ckanext.xloader.just_load_with_messytables
+      - key: ckanext.xloader.strict_type_guessing
+        default: True
+        example: False
+        description: |
+            Use with ckanext.xloader.use_type_guessing to set strict true or false
+            for type guessing. If set to False, the types will always fallback to string type.
```
Comment (ThrawnCA) [inline ckanext/xloader/config_declaration.yaml:56]: "fallback" -> "fall back"

Also, this should ideally have more description of what strictness means in this context.

File: ckanext/xloader/utils.py [hunk 2]
```diff
@@ -210,5 +214,7 @@ def type_guess(rows, types=TYPES, strict=False):
         # element in case of a tie
         # See: http://stackoverflow.com/a/6783101/214950
         guesses_tuples = [(t, guess[t]) for t in types if t in guess]
+        if not guesses_tuples:
```
Comment (ThrawnCA) [inline ckanext/xloader/utils.py:217]: How does this handle files with only a header and no data rows?

**Review by ThrawnCA — APPROVED** [review]

File: ckanext/xloader/utils.py (hunk 2 above)
Comment (duttonw) [reply ckanext/xloader/utils.py:217]: @ThrawnCA can you add a test for a header only upload please.

**Comment by ThrawnCA** [comment]
@JVickery-TBS Merge conflict needs to be resolved.

**Comment by JVickery-TBS** [comment]
@ThrawnCA fixed conflicts now

File: ckanext/xloader/utils.py (hunk 2 above)
Comment (ThrawnCA) [reply ckanext/xloader/utils.py:252]: @duttonw I can't; the pull request doesn't originate from our repo.

**Review by ThrawnCA — APPROVED** [review]

**Pull request merged**